packages/core/src/external/athenahealth/index.ts

```diff
@@ -397,6 +437,238 @@ class AthenaHealthApi {
+          });
+        }
+        return {
+          departmentid: this.stripDepartmentId(departmentId),
+          returnvitalsid: true,
+          source: "DEVICEGENERATED",
+          vitals: [vitalsData],
+          THIRDPARTYUSERNAME: undefined,
+          PATIENTFACINGCALL: undefined,
+        };
+      });
+      const settledResponses = await Promise.allSettled(
+        payloads.map(data =>
+          this.axiosInstanceProprietary.post(chartVitalsUrl, this.createDataParams(data))
+        )
+      );
+      const errors = settledResponses.flatMap(r => (r.status === "rejected" ? [r.reason] : []));
+      if (errors.length > 0) {
+        const msg = `Failure while creating some vitals @ AthenaHealth`;
+        log(`${msg}. Cause: ${errors.map(error => errorToString(error)).join(", ")}`);
+        capture.error(msg, {
+          extra: {
+            url: chartVitalsUrl,
+            cxId,
+            practiceId: this.practiceId,
+            patientId,
+            departmentId,
+            observationId: observation.id,
+            context: "athenahealth.create-vitals",
+            errors,
+          },
+        });
+      }
+      const successes = settledResponses
+        .flatMap(r => (r.status === "fulfilled" ? [r.value] : []))
+        .map(response => {
+          if (!response.data) throw new MetriportError(`No body returned from ${chartVitalsUrl}`);
+          debug(`${chartVitalsUrl} resp: `, () => JSON.stringify(response.data));
+          return response;
+        });
+      if (successes.length === 0) throw new MetriportError(`No vitals created`);
+      if (responsesBucket) {
+        const filePath = createHivePartitionFilePath({
+          cxId,
+          patientId,
+          date: new Date(),
+        });
+        const key = `athenahealth/chart/vitals/${filePath}/${uuidv7()}.json`;
+        this.s3Utils
+          .uploadFile({
+            bucket: responsesBucket,
+            key,
+            file: Buffer.from(
+              successes.map(response => JSON.stringify(response.data)).join("\n"),
+              "utf8"
+            ),
+            contentType: "application/json",
+          })
+          .catch(processAsyncError("Error saving to s3 @ AthenaHealth - createVitals"));
+      }
+      const outcomes = successes.map(response => vitalsCreateResponseSchema.parse(response.data));
+      if (!outcomes.every(outcome => outcome.success)) {
+        throw new MetriportError(`Vitals create not successful`);
+      }
+      return outcomes;
+    } catch (error) {
+      const msg = `Failure while creating some vitals @ AthenaHealth`;
+      log(`${msg}. Cause: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          url: chartVitalsUrl,
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          context: "athenahealth.create-vitals",
+          error,
+        },
+      });
+      throw error;
```

Comment:
ditto: same thing about duplicated Sentry events

Reply:
Will leave as is for now.